enterprise/internal/batches/store/site_credentials.go

```diff
@@ -0,0 +1,201 @@
+package store
+
+import (
+	"context"
+	"time"
+
+	"github.com/keegancsmith/sqlf"
+
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/database/dbutil"
+	"github.com/sourcegraph/sourcegraph/internal/extsvc/auth"
+)
+
+type SiteCredential struct {
+	ID                  int64
+	ExternalServiceType string
+	ExternalServiceID   string
+	Credential          auth.Authenticator
+	CreatedAt           time.Time
+	UpdatedAt           time.Time
+}
+
+func (s *Store) CreateSiteCredential(ctx context.Context, c *SiteCredential) error {
+	if c.CreatedAt.IsZero() {
+		c.CreatedAt = s.now()
+	}
+
+	if c.UpdatedAt.IsZero() {
+		c.UpdatedAt = c.CreatedAt
+	}
+
+	q := createSiteCredentialQuery(c)
+	return s.query(ctx, q, func(sc scanner) error {
+		return scanSiteCredential(c, sc)
+	})
+}
+
+var createSiteCredentialQueryFmtstr = `
+-- source: enterprise/internal/batches/store/site_credentials.go:CreateSiteCredential
+INSERT INTO
+	batch_changes_site_credentials (external_service_type, external_service_id, credential, created_at, updated_at)
+VALUES
+	(%s, %s, %s, %s, %s)
+RETURNING
+	%s
+`
+
+func createSiteCredentialQuery(c *SiteCredential) *sqlf.Query {
+	return sqlf.Sprintf(
+		createSiteCredentialQueryFmtstr,
+		c.ExternalServiceType,
+		c.ExternalServiceID,
+		&database.NullAuthenticator{A: &c.Credential},
+		c.CreatedAt,
+		c.UpdatedAt,
+		sqlf.Join(siteCredentialColumns, ","),
+	)
+}
+
+func (s *Store) DeleteSiteCredential(ctx context.Context, id int64) error {
+	res, err := s.ExecResult(ctx, deleteSiteCredentialQuery(id))
+	if err != nil {
+		return err
+	}
+
+	// Check the credential existed before.
+	if rows, err := res.RowsAffected(); err != nil {
+		return err
+	} else if rows == 0 {
+		return ErrNoResults
+	}
+	return nil
+}
+
+var deleteSiteCredentialQueryFmtstr = `
+-- source: enterprise/internal/batches/store/site_credentials.go:DeleteSiteCredential
+DELETE FROM
+	batch_changes_site_credentials
+WHERE
+	%s
+`
+
+func deleteSiteCredentialQuery(id int64) *sqlf.Query {
+	preds := []*sqlf.Query{
+		sqlf.Sprintf("id = %d", id),
+	}
+	return sqlf.Sprintf(
+		deleteSiteCredentialQueryFmtstr,
+		sqlf.Join(preds, "AND"),
+	)
+}
```

Comment:
I realise there's a standard pattern being used here, but I'm not sure this is clearer than:

```suggestion
func deleteSiteCredentialQuery(id int64) *sqlf.Query {
	return sqlf.Sprintf(
		deleteSiteCredentialQueryFmtstr,
		sqlf.Sprintf("id = %d", id),
	)
}
```

(or, indeed, hardcoding the `id = %d` into the query constant above)